### PR TITLE
Remove async wrappers completely. Utilize fake timers.

### PR DESCRIPTION
**Pull request by milesj**
An alternative to https://github.com/milesj/rut/pull/33

This new approach utilizes Jest's built in fake timers to simply exhaust them after rendering. With this, we can safely await async calls because all promises/timers simply happen within the same or next tick.

### CODE REVIEW

File: packages/jest-rut/src/integration.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  if (jest.getTimerCount() !== 0) {
+    jest.runAllImmediates();
+    jest.runAllTicks();
+    jest.runAllTimers();
+  }
+
+  if (!isFakeTimers) {
+    jest.useRealTimers();
+  }
```
Comment (alecklandgraf) [inline packages/jest-rut/src/integration.ts]: should this be `if (isFakeTimers)` ?

Comment (milesj) [reply packages/jest-rut/src/integration.ts]: This is to set back to real timers if the project is configured to use real timers and not fake ones.

File: packages/jest-rut/src/integration.ts [hunk 2]
```diff
@@ -9,11 +9,13 @@ export function runWithTimers<T>(cb: () => T): T {
   }
 
   const value = cb();
+  let loop = 0;
 
-  if (jest.getTimerCount() !== 0) {
+  while (jest.getTimerCount() > 0 && loop < 10) {
```
Comment (alecklandgraf) [inline packages/jest-rut/src/integration.ts]: would there be an edge case where a test has 11 timers/promises and this breaks?

File: packages/jest-rut/src/integration.ts (hunk 1 above)
Comment (alecklandgraf) [reply packages/jest-rut/src/integration.ts]: got it, thanks

File: packages/jest-rut/src/integration.ts (hunk 2 above)
Comment (milesj) [reply packages/jest-rut/src/integration.ts]: 11 timers wouldn't be an issue, since the `run` methods exhaust *all* timers. This is to simply avoid an infinite loop, caused by intervals or something similar.
